Scripts/travis.sh

```diff
@@ -19,22 +19,59 @@ set -euxo pipefail
 xcodebuild -version
 xcodebuild -showsdks
 
-if [ "${TYPE}" == "UNIT" ]; then
-  env NSUnbufferedIO=YES xcodebuild -project Tests/UnitTests/UnitTests.xcodeproj -scheme EarlGreyUnitTests -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO test | xcpretty -c;
-elif [ "${TYPE}" == "RUBY" ]; then
+# Runs xcodebuild retrying up to 3 times on failure to start testing (exit code 65).
+# The following arguments specified in the order below:
+#  $1 : .xcodeproj file
+#  $2 : scheme to run
+#
+# The output is prettified using xcpretty and redirected to xcodebuild.log for failure analysis.
+execute_xcodebuild() {
+  if [ -z ${1+x} ]; then
+    echo "first argument must be a valid .xcodeproj file"
+    exit 1
+  elif [ -z ${2+x} ]; then
+    echo "second argument must be a valid scheme"
+    exit 1
+  fi
+
+  retval_xcodebuild=0
+  for retry_attempts in {1..3}; do
+    # To retry on failure, disable exiting if command below fails.
+    set +e
+    env NSUnbufferedIO=YES xcodebuild -project ${1} -scheme ${2} -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO test | tee xcodebuild.log | xcpretty -s;
+    retval_xcodebuild=$?
+    # Even failed tests exit with code 65. Add a check to ensure test haven't started.
```

Comment:
nit: Either Aditya's comment or:
# Even failed tests exit with code 65, check to ensure tests haven't started.